docs/developer-guide/plugins.md

```diff
@@ -127,6 +127,25 @@ const ruleFunction = (primary, secondaryOptions) => {
 };
 ```
 
+## Quiet deprecation warnings
+
+We sometimes introduce deprecation warnings that impact plugin authors. You can individually quiet these in your plugin by handling the `emitWarning` event.
+
+For example, to quiet the deprecationg warning for `context.fix`:
+
+```js
+const original = process.emitWarning;
+process.emitWarning = function (message, options) {
+  if (
+    options?.type === "DeprecationWarning" &&
+    options?.code === "stylelint:005"
```

Comment:
[suggestion] Seeing `@types/node`, `emitWarning` supports several patterns for the arguments:

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/486660167625287171c6094685c0edac5600510e/types/node/process.d.ts#L790-L793

So, I suggest checking if `options` is an object or not. E.g.

```suggestion
    options &&
    typeof options === "object" &&
    options.type === "DeprecationWarning" &&
    options.code === "stylelint:005"
```